Python/pywarpx/_libwarpx.py

```diff
@@ -557,6 +561,165 @@ def get_particle_arrays(species_name, comp_name, level):
     _libc.free(data)
     return particle_data
 
+def get_particle_boundary_buffer(species_name, boundary, comp_name, level):
+    '''
+
+    This returns a list of numpy arrays containing the particle array data
+    for a species that has been scraped by a specific simulation boundary.
+
+    The data for the numpy arrays are not copied, but share the underlying
+    memory buffer with WarpX. The numpy arrays are fully writeable.
+
+    Parameters
+    ----------
+
+        species_name   : the species name that the data will be returned for.
+        boundary       : the boundary from which to get the scraped particle data.
+                         In the form x/y/z_hi/lo
+        comp_name      : the component of the array data that will be returned.
+                         If "step_scraped" the special attribute holding the
+                         timestep at which a particle was scraped will be
+                         returned.
+        level          : Which AMR level to retrieve scraped particle data from.
+    Returns
+    -------
+
+        A List of numpy arrays.
+
+    '''
+    if geometry_dim == '3d':
+        dimensions = {'x' : 1, 'y' : 2, 'z' : 3}
```

Comment:
Should this be `{'x' : 0, 'y' : 1, 'z' : 2}`?

```suggestion
        dimensions = {'x' : 0, 'y' : 1, 'z' : 2}
```

Reply:
Yeap, thanks! Hopefully this will now be fixed once and for all.

Reply:
This is fixed in an upstream commit that pulls this logic into another function to remove the code duplication, so I'll just cherry pick that here.